### DISCUSSION TITLE
Socket HostNotFound Errors do not trigger Enrich action?

### DISCUSSION THREAD

**Discussion by tl-amanda-oh** [open-telemetry/opentelemetry-dotnet#3208] [Q&A] [answered]
While implementing an enrich function, I noticed that HTTP Socket Exceptions with the error code `HostNotFound` do not trigger the enrich function on exception because of [these lines in `HttpHandlerDiagnosticListener`](https://github.com/open-telemetry/opentelemetry-dotnet/blob/2a97920ff0a603837a1121204824955bba739c57/src/OpenTelemetry.Instrumentation.Http/Implementation/HttpHandlerDiagnosticListener.cs#L218-L226).

This prevents additional information from being captured in the activity when the exception is a `HostNotFound` Socket Exception. 

Can someone shed some light on this particular behaviour? I could not find anything in the documentation.

Thank you.

**Comment by alanwest** [comment, marked as answer]
@tl-amanda-oh thanks for raising this. Seems reasonable to me that enrich should be invoked in this scenario. Opened an issue #3406.
